plugins/out_s3/s3.c

```diff
@@ -2383,6 +2404,14 @@ static struct flb_config_map config_map[] = {
      "key formatters all work as expected while this feature is set to true."
     },
 
+    {
+     FLB_CONFIG_MAP_STR, "storage_class", NULL,
+     0, FLB_FALSE, 0,
+     "Specify the storage class for S3 objects. This option can be enabled "
+     "when use_put_object is on. If this option is not specified, objects "
```

Comment:
If `use_put_object` is required, then you should add code in the init callback to check if that is configured and fail initialization if it isn't.

Reply:
@PettitWesley Thank you for taking your time.
I have updated pull request to check if use_put_object is enabled when storage_class is specified.